spec/ec2_wrapper_spec.rb

```diff
@@ -2,73 +2,119 @@
 
 describe Ec2Wrapper do
   
-  class TestWrapper
-    include Ec2Wrapper
+  def expect_wrapper
+    wrapper = Class.new do
+      include Ec2Wrapper
+    end.new
+    expect(wrapper).to receive(:ec2_client).and_return(
```

Comment:
I think you may want to use `allow(wrapper).to recieve(:ec2_client)` here instead of `expect(wrapper).to receive(:ec2_client)`.

Using `allow` sets up the mock object `wrapper` to receive the message `ec2_client`, but will not raise an exception of `wrapper.ec2_client` is not called.

Using `expect` also sets the mock object `wrapper` to receive the message `ec2_client`, but I think that it will then throw an exception unless `wrapper.ec2_client` is called.

In general, when I'm just setting up the mock, I use `allow(obj).to receive(:some_method)`. And then I use `expect(obj).to receive(:some_method)` only when I'm making an assertion that `:some_method` will definitely be called.